### PR TITLE
Push up to date

### PR DESCRIPTION
**Pull request by jgrund**
Add all commits up to master

### CODE REVIEW

**Comment by jgrund** [comment]
Looks like the bundles are being committed.

File: packages/srcmap-reverse/source/index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+export default () => {
+  server = http.createServer(
+    (request: http.IncomingMessage, response: http.ServerResponse) => {
+      const through = highland.pipeline(
+        highland.map(x => x.toString('utf-8')),
+        highland.collect(),
+        highland.map((x: string[]) => x.join('')),
+        highland.flatMap((x: string) => {
+          const { trace }: { trace: string } = JSON.parse(x);
```
Comment (jgrund) [inline packages/srcmap-reverse/source/index.js]: What is the full object coming out of here?

File: packages/srcmap-reverse/source/logger.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+const errorLog = logger({
+  path: 'srcmap-reverse-errors.log',
+  level: LEVELS.ERROR,
+  name: 'errors',
+  serializers: {}
+});
+
+const reverseLog = logger({
+  path: 'srcmap-reverse-trace.log',
```
Comment (jgrund) [inline packages/srcmap-reverse/source/logger.js]: Is this necessary?

File: packages/srcmap-reverse/source/reverse-in-parallel.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import highland from 'highland';
+import { exec } from 'child_process';
+import { errorLog } from './logger';
+
+export default (trace: string) => {
+  const lines = trace.split('\n');
+
+  return highland(lines)
+    .map((line: string) => {
+      return highland((push: Function) => {
```
Comment (jgrund) [inline packages/srcmap-reverse/source/reverse-in-parallel.js]: Why type this as Function?

File: packages/srcmap-reverse/source/reverse-in-parallel.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        const reverse = exec(
+          `node packages/reverser/dist/bundle.js`,
```
Comment (jgrund) [inline packages/srcmap-reverse/source/reverse-in-parallel.js]: This won't work this way in production.

File: packages/srcmap-reverse/package.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  "publishConfig": {
+    "access": "public"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/intel-hpdd/srcmap-reverse.git"
+  },
+  "author": "MFL Team",
+  "license": "MIT"
```
Comment (jgrund) [inline packages/srcmap-reverse/package.json]: This needs to depend on `@mfl/srcmap-reverser`

File: packages/srcmap-reverse/source/index.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import http from 'http';
+import highland from 'highland';
+import reverseInParallel from './reverse-in-parallel.js';
+
+let server: http.Server;
+
+export default () => {
+  server = http.createServer(
+    (request: http.IncomingMessage, response: http.ServerResponse) => {
+      const through = highland.pipeline(
```
Comment (johnsonw) [inline packages/srcmap-reverse/source/index.js]: Let's discuss this. I may need to add the type.

File: packages/srcmap-reverse/source/index.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        highland.map(x => x.toString('utf-8')),
+        highland.collect(),
+        highland.map((x: string[]) => x.join('')),
```
Comment (johnsonw) [inline packages/srcmap-reverse/source/index.js]: I don't think we do.

File: packages/srcmap-reverse/source/index.js (hunk 1 above)
Comment (johnsonw) [inline packages/srcmap-reverse/source/index.js]: It would be something like:
```
{
   trace: `Error: Come on sourcemaps.
at Object.DashboardFilterCtrl.$scope.filter.onFilterView (https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:38:7096)
at https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:14:4896
at https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:14:16407
at Scope.$eval (https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:12:12643)
at Scope.$apply (https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:12:12989)
at HTMLButtonElement.<anonymous> (https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:14:16389)
at HTMLButtonElement.jQuery.event.dispatch (https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:7:13226)
at HTMLButtonElement.elemData.handle (https://localhost:8000/static/chroma_ui/built-fd5ce21b.js:7:8056)`
}
```

File: packages/srcmap-reverse/source/logger.js (hunk 2 above)
Comment (johnsonw) [inline packages/srcmap-reverse/source/logger.js]: No, I wanted to discuss the logging a bit further. This was just something I added in while getting the logging setup.

File: packages/srcmap-reverse/source/reverse-in-parallel.js (hunk 3 above)
Comment (johnsonw) [inline packages/srcmap-reverse/source/reverse-in-parallel.js]: Updated to:
```
(e: ?Error, data: string) => void
```

File: packages/srcmap-reverse/source/reverse-in-parallel.js (hunk 4 above)
Comment (johnsonw) [inline packages/srcmap-reverse/source/reverse-in-parallel.js]: This is basically what we were doing in realtime. Let's discuss what needs to change.

File: package.json [hunk 8]
```diff
@@ -1,53 +1,63 @@
...
   },
-  "nyc": {
-    "sourceMap": false,
-    "instrument": false
-  },
-  "publishConfig": {
-    "registry": "https://ubit-artifactory-or.intel.com/artifactory/api/npm/iml-npm-local"
+  "config": {
+    "port": "8082",
+    "srcMapFile": "/test/fixtures/built-fd5ce21b.js.map.json"
```
Comment (jgrund) [inline package.json]: This being an absolute path seems strange.

File: packages/srcmap-reverse/package.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@mfl/srcmap-reverse",
+  "version": "2.0.0",
+  "description": "Reverses a minified stack trace using a source map.",
+  "main": "dist/bundle.js",
+  "scripts": {
+    "prepublish": "old_pwd=$PWD; cd ../..; yarn run rollup -- -i $old_pwd/source/index.js -o $old_pwd/$npm_package_main; chmod +x $old_pwd/$npm_package_main"
```
Comment (jgrund) [inline packages/srcmap-reverse/package.json]: Not sure you need the chmod here?

File: package.json (hunk 8 above)
Comment (jgrund) [inline package.json]: These also need to be on their relevant sub-packages. This file won't be there in production.

File: packages/srcmap-reverse/source/logger.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import { LEVELS, serializers, default as logger } from '@mfl/logger';
+
+const errorLog = logger({
+  path: 'srcmap-reverse-errors.log',
+  level: LEVELS.ERROR,
+  name: 'errors',
+  serializers
+});
+
+const reverseLog = logger({
```
Comment (jgrund) [inline packages/srcmap-reverse/source/logger.js]: This still needs the same name as the log we use in prod.

File: packages/srcmap-reverse/test/trace-reverse.integration.test.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+  afterEach((done: Function) => {
+    server.close(err => {
+      if (err) done.fail(err);
+      else done();
+    });
+  });
+
+  it('should return line and column numbers for each line in a minified stack trace', (
+    done: Function
```
Comment (jgrund) [inline packages/srcmap-reverse/test/trace-reverse.integration.test.js]: Why is this needed?

File: packages/srcmap-reverser/package.json [hunk 12]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@mfl/srcmap-reverser",
+  "version": "0.0.0",
+  "description": "Reverse a single line using a specified source map.",
+  "main": "dist/bundle.js",
+  "scripts": {
+    "prepublish": "old_pwd=$PWD; cd ../..; yarn run rollup -- -i $old_pwd/source/index.js -o $old_pwd/$npm_package_main; chmod +x $old_pwd/$npm_package_main"
```
Comment (jgrund) [inline packages/srcmap-reverser/package.json]: not sure we need the chmod?

File: packages/srcmap-reverser/source/index.js [hunk 13]
```diff
@@ -19,19 +21,12 @@
...
-    .map(x => traceItemRegex.exec(x))
-    .filter(x => x != null)
-    .map(xs => ({
-      compiledLine: xs[0],
-      url: xs[1],
-      line: parseInt(xs[2], 10),
-      column: parseInt(xs[3], 10)
-    }));
-};
+highland((process.stdin: Readable))
```
Comment (jgrund) [inline packages/srcmap-reverser/source/index.js]: Do we need to put Readable here?

File: packages/srcmap-reverser/source/logger.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+import { LEVELS, serializers, default as logger } from '@mfl/logger';
+
+const errorLog = logger({
+  path: 'srcmap-reverser-errors.log',
+  level: LEVELS.ERROR,
+  name: 'errors',
+  serializers
+});
+
+const reverseLog = logger({
```
Comment (jgrund) [inline packages/srcmap-reverser/source/logger.js]: Not sure we need this.

File: packages/srcmap-reverser/source/reverser.js [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+import type { HighlandStreamT } from 'highland';
+import type { Readable } from 'stream';
+
+const srcMapFile: string = process.env.npm_package_config_srcMapFile
+  ? process.env.npm_package_config_srcMapFile
+  : '';
+
+const sourceMapStream = highland(
+  (createReadStream(`${__dirname}/..${srcMapFile}`): Readable)
```
Comment (jgrund) [inline packages/srcmap-reverser/source/reverser.js]: I'm pretty sure we're going to want to provide an absolute path here.

File: packages/srcmap-reverser/source/src-map-reverse.js [hunk 16]
```diff
@@ -19,24 +21,32 @@
...
-export default (srcMap, trace) => {
-  var smc = new SourceMapConsumer(srcMap);
+export default (srcMap: string, trace: string) => {
+  const smc = new SourceMapConsumer(srcMap);
 
   return buildTraceCollection(trace)
-    .map(x => smc.originalPositionFor(x))
-    .map(function prepend (x) {
+    .map(smc.originalPositionFor.bind(smc))
+    .map((x: PositionData) => {
```
Comment (jgrund) [inline packages/srcmap-reverser/source/src-map-reverse.js]: Could be `tap`?

File: .npmignore [hunk 17]
```diff
@@ -6,6 +6,8 @@ node_modules
 .flowconfig
 .babelrc
 .gitignore
+yarn.lock
+*.log
 /test
-/source
 /coverage
```
Comment (johnsonw) [inline .npmignore]: added

File: package.json (hunk 8 above)
Comment (johnsonw) [inline package.json]: regarding the absolute path, i'll just make it "built-fd5ce21b.js.map.json"

File: packages/srcmap-reverse/package.json (hunk 9 above)
Comment (johnsonw) [inline packages/srcmap-reverse/package.json]: Perhaps not, I noticed it was in the device-scanner so I figured I would add it here as well.

File: packages/srcmap-reverser/package.json (hunk 12 above)
Comment (johnsonw) [inline packages/srcmap-reverser/package.json]: removed

File: packages/srcmap-reverser/source/index.js (hunk 13 above)
Comment (johnsonw) [inline packages/srcmap-reverser/source/index.js]: Looks like we can get rid of it now.

File: packages/srcmap-reverser/source/src-map-reverse.js (hunk 16 above)
Comment (johnsonw) [inline packages/srcmap-reverser/source/src-map-reverse.js]: It's working over an array so tap isn't available.

File: packages/srcmap-reverse/source/reverse-in-parallel.js [hunk 18]
```diff
@@ -25,16 +25,14 @@ import highland from 'highland';
...
 export default (trace: string) => {
   const lines = trace.split('\n');
 
   return highland(lines)
     .map((line: string) => {
-      return highland((push: pushFn<Buffer>): void => {
+      return highland(push => {
         const reverse = exec(
-          `node packages/srcmap-reverser/dist/bundle.js`,
+          `node ${__dirname}/../node_modules/@mfl/srcmap-reverser/dist/bundle.js`,
```
Comment (jgrund) [inline packages/srcmap-reverse/source/reverse-in-parallel.js]: I think you want something like require.resolve here. We don't know where this module is going to live in prod.

File: packages/srcmap-reverser/source/reverser.js [hunk 19]
```diff
@@ -26,15 +26,12 @@ import srcmapReverse from './src-map-reverse.js';
 import highland from 'highland';
 
 import type { HighlandStreamT } from 'highland';
-import type { Readable } from 'stream';
 
-const srcMapFile: string = process.env.npm_package_config_srcMapFile
-  ? process.env.npm_package_config_srcMapFile
-  : '';
+const srcMapFile: string = process.env.NODE_ENV === 'test'
```
Comment (jgrund) [inline packages/srcmap-reverser/source/reverser.js]: This seems like the wrong way to go about this.

File: packages/srcmap-reverse/source/logger.js [hunk 20]
```diff
@@ -23,18 +23,13 @@
 
 import { LEVELS, serializers, default as logger } from '@mfl/logger';
 
-const errorLog = logger({
-  path: 'srcmap-reverse-errors.log',
+const clientErrorsLog = logger({
```
Comment (jgrund) [inline packages/srcmap-reverse/source/logger.js]: Can't we just export here?

File: packages/srcmap-reverse/source/logger.js [hunk 21]
```diff
@@ -23,18 +23,13 @@
...
+  path: process.env.NODE_ENV === 'test'
```
Comment (jgrund) [inline packages/srcmap-reverse/source/logger.js]: This seems like the wrong way to go about this.

File: packages/srcmap-reverse/source/reverse-in-parallel.js [hunk 22]
```diff
@@ -35,7 +35,7 @@ export default (trace: string) => {
           `node ${__dirname}/../node_modules/@mfl/srcmap-reverser/dist/bundle.js`,
           (err, x) => {
             if (err) {
-              errorLog.error({ err }, 'Reversing source map');
+              clientErrorsLog.error({ err }, 'Reversing source map');
```
Comment (jgrund) [inline packages/srcmap-reverse/source/reverse-in-parallel.js]: I think we only want to use the client-errors.log for successful traces.
